irods_specific_implementation/irods_rest_list_api_implementation.h

```diff
@@ -71,36 +72,69 @@ namespace irods::rest {
                     objects.push_back(obj_info);
                 }
                 else if (fcli::is_collection(*conn(), start_path)) {
-                    for (auto&& p : fcli::recursive_collection_iterator(*conn(), start_path)) {
-                        // skip earlier entries for paging
-                        if (offset > 0 && offset_counter < offset) {
-                            ++offset_counter;
-                            continue;
-                        }
-
-                        try {
-                            const auto object_status = fcli::status(*conn(), p.path());
-
-                            nlohmann::json obj_info = nlohmann::json::object();
-                            obj_info["type"] = type_to_string.at(object_status.type());
-                            obj_info["logical_path"] = p.path().c_str();
-
-                            if (stat) { aggregate_stat_information(*conn(), obj_info, p.path()); }
-                            if (permissions) { aggregate_permissions_information(obj_info, object_status.permissions()); }
-                            if (metadata) { aggregate_metadata_information(*conn(), obj_info, p.path()); }
-
-                            objects.push_back(obj_info);
-                        }
-                        catch (const irods::exception& e) {
-                            error("Caught exception - [error_code={}] {}", e.code(), e.what());
-                            return make_error_response(e.code(), e.client_display_what());
-                        }
-
-                        ++limit_counter;
-                        if (limit > 0 && limit_counter >= limit) {
-                            break;
-                        }
-                    } // for path
+		    if ("1" == _recursive) {
+			for (auto&& p : fcli::recursive_collection_iterator(*conn(), start_path)) {
+			    // skip earlier entries for paging
+			    if (offset > 0 && offset_counter < offset) {
+				++offset_counter;
+				continue;
+			    }
+
+			    try {
+				const auto object_status = fcli::status(*conn(), p.path());
+
+				nlohmann::json obj_info = nlohmann::json::object();
+				obj_info["type"] = type_to_string.at(object_status.type());
+				obj_info["logical_path"] = p.path().c_str();
+
+				if (stat) { aggregate_stat_information(*conn(), obj_info, p.path()); }
+				if (permissions) { aggregate_permissions_information(obj_info, object_status.permissions()); }
+				if (metadata) { aggregate_metadata_information(*conn(), obj_info, p.path()); }
+
+				objects.push_back(obj_info);
+			    }
+			    catch (const irods::exception& e) {
+				error("Caught exception - [error_code={}] {}", e.code(), e.what());
+				return make_error_response(e.code(), e.client_display_what());
+			    }
+
+			    ++limit_counter;
+			    if (limit > 0 && limit_counter >= limit) {
+				break;
+			    }
+			} // for path
+		    } else {
```

Comment:
this new code block is identical to the recursive block... can we just make p dependent on the recursive flag?   and then only have this code block once?

happy if the answer is, yes, but later.  if so, please make an issue for refactor/consolidate.

Reply:
Definitely can think about the cleanest way to do that.

Reply:
+1

Reply:
Done